rs/sns/swap/src/swap.rs

```diff
@@ -1618,155 +1621,42 @@ impl Swap {
-                        Some(p) => p,
-                        // principal_str should always be parseable as a PrincipalId as that is enforced
-                        // in `refresh_buyer_tokens`. In the case of a bug due to programmer error, increment
-                        // the invalid field. This will require a manual intervention via an upgrade to correct
-                        None => {
-                            sweep_result.invalid += 1;
-                            continue;
-                        }
-                    };
-
-                    (None, parsed_buyer_principal, None)
-                }
-                Some(Investor::CommunityFund(cf_investment)) => {
-                    let nf_neuron_nns_controller = match cf_investment.try_get_controller() {
-                        Ok(controller) => controller,
-                        Err(e) => {
-                            log!(ERROR, "Invalid NF neuron: recipe={recipe:?} error={e}");
-                            sweep_result.invalid += 1;
-                            continue;
-                        }
-                    };
-                    (
-                        Some(nf_neuron_nns_controller),
-                        PrincipalId::from(nns_governance),
-                        Some(cf_investment.nns_neuron_id),
-                    )
-                }
-                // SnsNeuronRecipe.investor should always be present as it is set in `commit`.
-                // In the case of a bug due to programmer error, increment the invalid field.
-                // This will require a manual intervention via an upgrade to correct
-                None => {
-                    log!(
-                        ERROR,
-                        "Missing investor information for neuron recipe {:?}",
-                        recipe,
-                    );
-                    sweep_result.invalid += 1;
-                    continue;
-                }
-            };
-
-            let (dissolve_delay_seconds, memo, followees) = match recipe.neuron_attributes.as_ref()
-            {
-                Some(neuron_attribute) => (
-                    neuron_attribute.dissolve_delay_seconds,
-                    neuron_attribute.memo,
-                    neuron_attribute.followees.clone(),
-                ),
-                // SnsNeuronRecipe.neuron_attributes should always be present as it is set in `commit`.
-                // In the case of a bug due to programmer error, increment the invalid field.
-                // This will require a manual intervention via an upgrade to correct
-                None => {
-                    log!(
-                        ERROR,
-                        "Missing neuron_attributes information for neuron recipe {:?}",
+            // Here we convert the SnsNeuronRecipe (a Swap concept) to an SnsNeuronRecipe (an SNS Governance concept)
+            match recipe.to_neuron_recipe(nns_governance, sns_transaction_fee_e8s) {
```

Comment:
NCR;

I think I would still slightly prefer something like 

```
sns_gov::SnsNeuronRecipe::from((recipe, nns_governance, sns_transaction_fee_e8s))
```

but I'm okay to keep as-is, so long as we have comments explaining the intention.

Reply:
Unfortunately, this can't work due to Rust not allowing implementing traits across crates.